### PR TITLE
Fixes #18237: cli rework: doc

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18237

### CODE REVIEW

**Comment by peckpeck** [comment]
This pr needs a rebase

**Comment by gpoblon** [comment]
PR rebased

**Comment by gpoblon** [comment]
Commit modified

File: rudder-lang/docs/usage.adoc [hunk 1]
```diff
@@ -14,65 +14,157 @@ Optionally add rudderc to your path `export PATH=$PATH:/opt/rudder/bin/rudderc`
...
 === Usage
 
-`rudderc` has 2 features:
+`rudderc` has 4 features, called actions that will generate code. The action you need fully depend on the format you have and the output format you want:
 
-* Compile which is the default behavior: takes a _RL_ technique as an input and generates the desired 
-* Translate which is the alternative mode:  takes a _JSON_ technique as an input and generates a _rudder-lang_ technique (_.rl_) 
+* compile: generates either a DSC / CFEngine technique from a RudderLang technique
+* migrate: generates a RudderLang technique from a CFEngine technique
+* technique read: generates a JSON technique from a RudderLang technique
```
Comment (peckpeck) [inline rudder-lang/docs/usage.adoc:21]: add that this is a json in the technique editor format

File: rudder-lang/docs/usage.adoc [hunk 2]
```diff
@@ -14,65 +14,157 @@ Optionally add rudderc to your path `export PATH=$PATH:/opt/rudder/bin/rudderc`
...
+* technique generate: generates a JSON object that comes with RudderLang + DSC + CFEngine technique from a JSON technique. Please note that by exception, `--stdin` and `--stdout` options are default behavior for `technique generate`.
```
Comment (peckpeck) [inline rudder-lang/docs/usage.adoc:22]: i think technique read should also use stdin by default

File: rudder-lang/docs/usage.adoc [hunk 3]
```diff
@@ -14,65 +14,157 @@ Optionally add rudderc to your path `export PATH=$PATH:/opt/rudder/bin/rudderc`
...
 
-Any information log is displayed to `SDTOUT`, errors should be printed to `STDERR`
+Any information log is displayed to `SDTOUT`, errors should be printed to `STDERR`. JSON log format handled. JSON output (including logs) is optional for compile and migrate but is default behavior for `technique read` and `generate`.
```
Comment (peckpeck) [inline rudder-lang/docs/usage.adoc:24]: This is not easy to understand, could you rephrase this ?

File: rudder-lang/docs/usage.adoc [hunk 4]
```diff
@@ -14,65 +14,157 @@ Optionally add rudderc to your path `export PATH=$PATH:/opt/rudder/bin/rudderc`
...
 
 [align=center]
 image::rudderc-simple.svg[rudderc abilities]
 
 .The CLI usage (`rudderc --help` or `rudderc -h` output slightly modified)
 ----
+Rudderc abilities are callable through subcommands, namely <technique read>, <technique generate>, <migrate>, <compile>,
```
Comment (peckpeck) [inline rudder-lang/docs/usage.adoc:31]: abilities ?

File: rudder-lang/docs/usage.adoc [hunk 5]
```diff
@@ -14,65 +14,157 @@ Optionally add rudderc to your path `export PATH=$PATH:/opt/rudder/bin/rudderc`
...
+_rudderc_ abilities / subcommands are called *actions*, that all share several flags and options
+.Shared FLAGS / OPTIONS:
+----
 FLAGS:
+    -b, --backtrace    Generates a backtrace in case an error occurs
     -h, --help         Prints help information
-    -j, --json-log     Use json logs instead of human readable output
-    -t, --translate    Use technique translation mode rather than default compilation mode
+        --stdin        Takes stdin as an input rather than using a file. Overwrites input file option
+        --stdout       Takes stdout as an output rather than using a file. Overwrites output file option. Dismiss logs, including errors
```
Comment (peckpeck) [inline rudder-lang/docs/usage.adoc:63]: errors are on stderr why are they dismissed ?

**Review by peckpeck — CHANGES_REQUESTED** [review]

**Comment by gpoblon** [comment]
PR updated with a new commit

**Review by peckpeck — APPROVED** [review]

**Comment by gpoblon** [comment]
OK, squash merging this PR
